src/Halogen/VDom/Driver.purs

```diff
@@ -153,26 +241,56 @@ renderSpec document container =
      . (Input action -> Effect Unit)
     -> (ComponentSlotBox slots Aff action -> Effect (RenderStateX RenderState))
     -> HTML (ComponentSlot slots Aff action) action
+    -> Boolean
     -> Maybe (RenderState state action slots output)
     -> Effect (RenderState state action slots output)
-  render handler child (HTML vdom) =
+  render handler renderChild (HTML vdom) isRoot =
     case _ of
       Nothing -> do
-        renderChildRef <- Ref.new child
-        let spec = mkSpec handler renderChildRef document
+        traceM { message: "Halogen.VDom.Driver.renderSpec render", handler, renderChild, vdom, container }
+        renderChildRef <- Ref.new renderChild
+        let renderChildHydrate = \_componentSlotBox _element -> throw "should not be called" -- TODO!!!!!
```

Comment:
Do you mind explaining this TODO?

Reply:
To reduce number of possible errors / increase totality I was thinking about splitting vdomSpec (currently has rendering and hydration functionlity) to vdomSpec (only render) and hydrateVdomSpec (render, because it's used after hydration, and hydrate functionality)